frame/beefy/src/lib.rs

```diff
@@ -154,11 +162,25 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 		I: Iterator<Item = (&'a T::AccountId, T::BeefyId)>,
 	{
 		let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
+		let bounded_next_authorities = WeakBoundedVec::<_, T::MaxAuthorities>::force_from(
+			next_authorities,
+			Some(
+				"Warning: The session has more validators than expected. \
+				A runtime configuration adjustment may be needed.",
+			),
+		);
```

Comment:
I think instead we should truncate the authorities list to fit, and then put a log message

Reply:
Or even update this API to accept a bounded vec to start with (if possible), so there will be no issue with passing in a value with the wrong size.